src/tf/tf.cpp

```diff
@@ -339,6 +340,62 @@ struct tf_parser
         return prog.add_instruction(op, {args[0], weights});
     }
 
+    instruction_ref parse_depthwiseconv(const std::string&,
+                                        attribute_map attributes,
+                                        std::vector<instruction_ref> args)
+    {
+        op::convolution op;
+        size_t num_channels = args[0]->get_shape().lens()[1];
+        op.group            = num_channels;
+        if(contains(attributes, "padding"))
+        {
+            const std::string& pad_mode = attributes.at("padding").s();
+            if(pad_mode.find("SAME") != std::string::npos)
+            {
+                op.padding_mode = op::padding_mode_t::same;
+            }
+        }
+        if(contains(attributes, "strides"))
+        {
+            std::vector<size_t> stride;
+            copy(attributes.at("strides").list().i(), std::back_inserter(stride));
+            reorder_data(stride);
+            if(stride.size() != 4)
+            {
+                MIGRAPHX_THROW("strides should have 4 values");
+            }
+            op.stride[0] = stride[2];
+            op.stride[1] = stride[3];
+        }
+        auto weights = args[1];
+        // check if weights are from a constant
+        if(weights->name() != "@param")
+        {
+            if(is_nhwc)
+            {
+                weights = prog.add_instruction(op::transpose{{1, 3, 0, 2}}, args[1]);
```

Comment:
If weights is literal, you can transpose it into a new literal, then transpose instruction does not needed to be inserted into the program.

Reply:
Deleted previous comment by accident, but the jist was that the weights have a different format: [out_channels, in_channels, h, w] compared to NCHW vs NHWC. I insert this additional transpose to get the literal into the correct format. The simplify_reshapes pass should replace the transposes with the correct reshape as well.

Reply:
As long as the transpose is removed, I will have less concern.

Reply:
It will be removed by the const propogation pass.